### PR TITLE
agent-ui: use redux-persist

### PR DESCRIPTION
**Pull request by jeremie-stratumn**
https://github.com/stratumn/indigo-js/issues/74

---
This change is [Reviewable](https://reviewable.io/reviews/stratumn/indigo-js/82)

### CODE REVIEW

File: packages/agent-ui/src/containers/index.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * They are tightly coupled to the app ("smart components").
  * See http://redux.js.org/docs/basics/UsageWithReact.html
  */
-export { default as Root } from './root';
+// export { default as Root } from './root';
```
Comment (jeremie-stratumn) [inline packages/agent-ui/src/containers/index.js]: Jest does not like the `import { PersistGate } from 'redux-persist/es/integration/react';` in this file.. I could not find why online, so I just did this workaround. I will spend more time trying to figure this out, not to be merged as is.

Comment (t-bast) [reply packages/agent-ui/src/containers/index.js]: This is very strange indeed... :/

File: packages/agent-ui/src/store/configure.js [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
 
 /* It's ok (even encouraged) to include the Redux dev tools in production:
  * https://medium.com/@zalmoxis/using-redux-devtools-in-production-4c5b56c5600f
  * If we need to improve perf, we can use redux-devtools-extension/logOnlyInProduction
  */
-const configureStore = () =>
-  createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
+
+const config = {
+  key: 'abc',
```
Comment (t-bast) [inline packages/agent-ui/src/store/configure.js]: ?

File: packages/agent-ui/src/store/configure.js [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
+  storage,
+  // only persist the agents part
+  whitelist: ['agents']
+};
+
+const reducer = persistReducer(config, rootReducer);
```
Comment (t-bast) [inline packages/agent-ui/src/store/configure.js]: nit: the name could be more meaningful than "reducer" :)
Something like rootReducerWithPersistence?

File: packages/agent-ui/src/store/configure.js (hunk 2 above)
Comment (jeremie-stratumn) [reply packages/agent-ui/src/store/configure.js]: yeah, i will change this to a more explicit key, like `stratumn-agent-ui`

File: packages/agent-ui/src/store/configure.js (hunk 3 above)
Comment (jeremie-stratumn) [reply packages/agent-ui/src/store/configure.js]: or `persistentReducer`. will change it

**Comment by t-bast** [comment]
Cool, good to know that we don't need a specific action / reducer, thanks for checking !

**Comment by t-bast** [comment]
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/82)*

**Comment by t-bast** [comment]
:lgtm:

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/82#-:-KxnBBaThFVbl0nIzuiD:bnfp4nl)*

**Comment by jeremie-stratumn** [comment]
I've made the changes requested + fixed the import issue thanks to @such. I have not figured out a way to test it properly though... Will do that later!

**Pull request merged**